packages/ic-asset-certification/tests/large_assets.rs

```diff
@@ -0,0 +1,193 @@
+    let chunk_one_res = asset_router
+        .serve_asset(&cbor_encoded_certificate, &chunk_one_req)
+        .unwrap();
+    assert_response_eq!(chunk_one_res, expected_chunk_one_res);
+
+    let chunk_one_verification = verify_request_response_pair(
+        chunk_one_req,
+        chunk_one_res,
+        canister_id.as_ref(),
+        current_time,
+        MAX_CERT_TIME_OFFSET_NS,
+        &root_key,
+        MIN_REQUESTED_VERIFICATION_VERSION,
+    )
+    .unwrap();
+    assert_eq!(chunk_one_verification.verification_version, 2);
+    assert_verified_response_eq!(
+        chunk_one_verification.response.unwrap(),
+        expected_chunk_one_res
+    );
+
+    let mut expected_headers = common_asset_headers();
+    expected_headers.extend(vec![
+        ("content-type".to_string(), "text/html".to_string()),
+        (
+            "content-length".to_string(),
+            (ASSET_ONE_SIZE - ASSET_CHUNK_SIZE).to_string(),
+        ),
+        (
+            "content-range".to_string(),
+            format!(
+                "bytes {}-{}/{}",
+                ASSET_CHUNK_SIZE,
+                ASSET_ONE_SIZE - 1,
+                ASSET_ONE_SIZE
+            ),
+        ),
+    ]);
+    let expected_chunk_two_res = HttpResponse::builder()
+        .with_status_code(StatusCode::PARTIAL_CONTENT)
+        .with_headers(expected_headers)
+        .with_body(asset_one_chunk_two)
+        .build();
+
+    let chunk_two_req = HttpRequest::get(&req_url)
+        .with_headers(vec![(
+            "range".to_string(),
+            format!("bytes={}-", ASSET_CHUNK_SIZE),
+        )])
+        .build();
+    let chunk_two_res = asset_router
+        .serve_asset(&cbor_encoded_certificate, &chunk_two_req)
+        .unwrap();
+    assert_response_eq!(chunk_two_res, expected_chunk_two_res);
+
+    let chunk_two_verification = verify_request_response_pair(
+        chunk_two_req,
+        chunk_two_res,
+        canister_id.as_ref(),
+        current_time,
+        MAX_CERT_TIME_OFFSET_NS,
+        &root_key,
+        MIN_REQUESTED_VERIFICATION_VERSION,
+    )
+    .unwrap();
+    assert_eq!(chunk_two_verification.verification_version, 2);
+    assert_verified_response_eq!(
+        chunk_two_verification.response.unwrap(),
+        expected_chunk_two_res
+    );
+}
+
+#[fixture]
+fn asset_cel_expr() -> String {
+    DefaultCelBuilder::full_certification()
+        .with_response_certification(DefaultResponseCertification::response_header_exclusions(
+            vec![],
+        ))
+        .build()
+        .to_string()
+}
+
+#[fixture]
+fn asset_range_cel_expr() -> String {
+    DefaultCelBuilder::full_certification()
+        .with_request_headers(vec!["range"])
+        .with_response_certification(DefaultResponseCertification::response_header_exclusions(
+            vec![],
+        ))
+        .build()
+        .to_string()
+}
+
+#[fixture]
+fn asset_one_body() -> &'static Vec<u8> {
+    static ASSET_ONE_BODY: OnceCell<Vec<u8>> = OnceCell::new();
```

Comment:
Another experiment here, to try and improve the performance of tests with large assets. In this case it only makes a minor difference, but it should make a bigger difference for tests with larger files, especially when we reuse the same assets across multiple tests.